Source/santad/EventProviders/SNTEndpointSecurityManager.mm

```diff
@@ -0,0 +1,304 @@
+      sm.action = ACTION_NOTIFY_WRITE;
+      sm.ppid = m->process->original_ppid;
+      strncpy(sm.path, m->event.close.target->path.data, m->event.close.target->path.length);
+      sm.path[m->event.close.target->path.length] = '\0';
+      callback = self.logCallback;
+      audit_token = m->process->audit_token;
+      break;
+    }
+    case ES_EVENT_TYPE_NOTIFY_UNLINK:
+      sm.action = ACTION_NOTIFY_DELETE;
+      sm.ppid = m->process->original_ppid;
+      strncpy(sm.path, m->event.unlink.target->path.data, m->event.unlink.target->path.length);
+      sm.path[m->event.unlink.target->path.length] = '\0';
+      callback = self.logCallback;
+      audit_token = m->process->audit_token;
+    case ES_EVENT_TYPE_NOTIFY_TRUNCATE: {
+      sm.action = ACTION_NOTIFY_DELETE;
+      sm.ppid = m->process->original_ppid;
+      strncpy(sm.path, m->event.truncate.target->path.data, m->event.truncate.target->path.length);
+      sm.path[m->event.truncate.target->path.length] = '\0';
+      callback = self.logCallback;
+      audit_token = m->process->audit_token;
+      break;
+    }
+    case ES_EVENT_TYPE_NOTIFY_LINK: {
+      sm.action = ACTION_NOTIFY_LINK;
+      sm.ppid = m->process->original_ppid;
+      strncpy(sm.path, m->event.link.source->path.data, m->event.link.source->path.length);
+      sm.path[m->event.link.source->path.length] = '\0';
+      strncpy(sm.newpath, m->event.link.target_filename.data, m->event.link.target_filename.length);
+      sm.newpath[m->event.link.target_filename.length] = '\0';
+      callback = self.logCallback;
+      audit_token = m->process->audit_token;
+      break;
+    }
+    case ES_EVENT_TYPE_NOTIFY_RENAME: {
+      sm.action = ACTION_NOTIFY_RENAME;
+      sm.ppid = m->process->original_ppid;
+      strncpy(sm.path, m->event.rename.source->path.data, m->event.rename.source->path.length);
+      sm.path[m->event.rename.source->path.length] = '\0';
+
+      switch(m->event.rename.destination_type) {
+        case ES_DESTINATION_TYPE_NEW_PATH:
+          strncpy(sm.newpath, m->event.rename.destination.new_path.filename.data, m->event.rename.destination.new_path.filename.length);
+          sm.newpath[m->event.rename.destination.new_path.filename.length] = '\0';
+          break;
+        case ES_DESTINATION_TYPE_EXISTING_FILE:
+          strncpy(sm.newpath, m->event.rename.destination.existing_file->path.data, m->event.rename.destination.existing_file->path.length);
+          sm.newpath[m->event.rename.destination.existing_file->path.length] = '\0';
+          break;
+      }
+
+      callback = self.logCallback;
+      audit_token = m->process->audit_token;
+    }
+    default:
+      break;
+  }
+
+  if (self.prefixTree->HasPrefix(sm.path)) {
+    return;
+  }
+
+  if (callback) {
+    sm.uid = audit_token_to_ruid(audit_token);
+    sm.gid = audit_token_to_rgid(audit_token);
+    sm.pid = audit_token_to_pid(audit_token);
+    proc_name(sm.pid, sm.pname, 1024);
```

Comment:
I think this gets truncated to MAXCOMLEN ~32 Bytes.
Does `[NSRunningApplication runningApplicationWithProcessIdentifier:sm.pid].localizedName` work at this stage?

Reply:
That seems to be less reliable than proc_name and proc_name is what 'old' Santa used.